plugins/entity_filter.py

```diff
@@ -355,46 +427,28 @@ def entity_filter_task(self, db_id: int) -> str:
-    output_entities_random_rows = []
-    sampling_counter = 0
-    mimetype = mimetypes.MimeTypes().guess_type(input_file_url)[0]
-    with open_url(input_file_url, stream=True) as url_data:
-        for entity in ensure_dict(load_entities(file_=url_data, mimetype=mimetype)):
-            if entity["ID"] in id_list:
-                # find entities in id_list if id_list not empty
-                output_entities_id_list.append(entity) # TODO
-                id_list.remove(entity["ID"])
-
-            elif n_sampled_rows > 0:
-                # sample rows to fill up remaining rows according to row sampling
-                if row_sampling == RowSamplingType.RANDOM.value:
-                    if sampling_counter < n_sampled_rows:
-                        # add first n
-                        output_entities_random_rows.append(entity)
-                        sampling_counter += 1
-                    else:
-                        # add with prob n/(n+1) at random index
-                        if random.random() < n_sampled_rows/(sampling_counter + 1):
-                            
-                            index = random.randrange(n_sampled_rows)
-                            output_entities_random_rows[index] = entity
-                        sampling_counter += 1
-                elif row_sampling == RowSamplingType.FIRST_N.value:
-                    if sampling_counter < n_sampled_rows:
-                        output_entities_random_rows.append(entity)
-                        sampling_counter += 1
-                else:
-                    raise ValueError("Invalid argument for Row Sampling!")
-
-        if id_list: # not all ID's in file
-            raise ValueError(f"The following ID's could not be found: {str(id_list)}")
-            # TODO: do sth else?
+    
+    
+    # mimetype of input and output file
+    mimetype : str = None
+    # list of output entities
+    output_entities : Dict[str, Any] = {}
+    if n_rows > 0:
+        with open_url(input_file_url, stream=True) as url_data:
+            try:
+                mimetype = url_data.headers['Content-Type']
+            except:
+                mimetype = mimetypes.MimeTypes().guess_type(input_file_url)[0]
+
+            output_entities = filter_rows(url_data=url_data, mimetype=mimetype, id_list=id_list, 
+                                          n_sampled_rows=n_sampled_rows, row_sampling=row_sampling)
+            if len(output_entities) != n_rows:
+                raise ValueError("Number of rows requested is greater than number of rows in input file!")
+                # TODO: maybe log an error instead of exception
 
-    output_entities = output_entities_id_list + output_entities_random_rows
-    if len(output_entities) != n_rows:
-        raise ValueError("Number of rows requested is greater than number of rows in input file!")
+    else:
+        # TODO take all rows
```

Comment:
It should be easy to switch off sampling and use INFINITY for n_rows to implement this case with minimal code changes.
Alternatively make sure that the second part of the pipeline (filtering columns) can work on streams/iterables and just use the entities iterable as is without row filtering.